cpp/merkletree/sparse_merkle_tree.h

```diff
@@ -171,29 +172,25 @@ class SparseMerkleTree {
   // or the leaf index is 0.
   //
   // @param path the path of the leaf whose inclusion proof to return.
-  std::vector<std::string> InclusionProof(const Path& path);
+  std::vector<std::string> InclusionProof(const Path& path) const;
 
   std::string Dump() const;
 
  private:
-  // WARNING WARNING WARNING
-  // 64 < 256 !
-  // WARNING WARNING WARNING
-  // TODO(alcutter): BIGNUM probably.
-  typedef uint64_t IndexType;
+  typedef cert_trans::BigNum IndexType;
```

Comment:
Part of the reason was to take advantage of the expand-to-fit nature of bignums, and I must admit it didn't occur to me that a freshly init'd BIGNUM structure could possibly be 24 bytes in size.  Back to the drawing board.